src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -538,8 +538,60 @@
 
    if (length(pieces) > 1)
       print(help(pieces[2], package=pieces[1], help_type='html'))
+   else {
+      # try custom help handler, otherwise fall through to default handler
+      helpUrl <- .rs.getCustomHelpUrl(token)
+      if (!is.null(helpUrl)) {
+         if (nzchar(helpUrl)) # handlers return "" to indicate no help available
+            utils::browseURL(helpUrl)
+      } else {
+         print(help(pieces[1], help_type='html', try.all.packages=TRUE))
+      }
+   }
+})
+
+.rs.addFunction("findCustomHelpContext", function(token, handler) {
+   
+   # if the token has a '$' in it then it might have a custom
+   # help handler that can field this request
+   if (grepl("\\$", token)) {
+      
+      # split on $ (it has at least one so components will be > 1)
+      components <- strsplit(token, "\\$")[[1]]
+      topic <- components[[length(components)]]
+      source <- paste(components[1:length(components)-1], collapse = "$")
```

Comment:
I think precedence rules are biting you here -- this should probably be

    1:(length(components)-1)

as otherwise you're generating a vector ranging from `0` to `length(components)-1` inclusive.